lib/zlib.js

```diff
@@ -275,7 +275,7 @@ function ZlibBase(opts, mode, handle, { flush, finishFlush, fullFlush }) {
   this._defaultFlushFlag = flush;
   this._finishFlushFlag = finishFlush;
   this._defaultFullFlushFlag = fullFlush;
-  this.once('end', this.close);
+  this.once('end', _close.bind(this, this));
```

Comment:
```suggestion
  this.once('end', _close.bind(null, this));
```
(One less reference for the GC to contend with.)